nextjs-app/src/pages/form.tsx

```diff
@@ -67,42 +68,47 @@ export default function Form() {
                     onInput={handleUsernameInput as any}
                     type="text"></PieInput>
 
-                <label htmlFor="email">
+                <PieFormLabel for="email">
                     Email:
-                </label>
-                <input
+                </PieFormLabel>
+                <PieInput
                     className="form-field"
                     id="email"
                     data-test-id="email"
                     name="email"
-                    onChange={handleEmailChange}
-                    type="email" />
+                    value={email}
+                    onInput={handleEmailChange as any}
+                    type="email"></PieInput>
                 
-                <label htmlFor="password">
+                <PieFormLabel for="password">
                     Password:
-                </label>
-                <input
+                </PieFormLabel>
+                <PieInput
                     className="form-field"
                     id="password"
                     data-test-id="password"
                     name="password"
-                    onChange={handlePasswordChange}
-                    type="password" />
+                    value={password}
+                    onInput={handlePasswordChange as any}
+                    type="password"></PieInput>
 
-                <label htmlFor="passwordConfirmation">
+                <PieFormLabel for="passwordConfirmation">
                     Confirm Password:
-                </label>
-                <input
+                </PieFormLabel>
+                <PieInput
                     className="form-field"
                     id="passwordConfirmation"
                     data-test-id="passwordConfirmation"
                     name="passwordConfirmation"
-                    onChange={handlePasswordConfirmationChange}
-                    type="password" />
+                    value={passwordConfirmation}
+                    onInput={handlePasswordConfirmationChange as any}
+                    type="password"></PieInput>
 
                 <div className="form-controls">
+                    <PieFormLabel for="approveSettings">
+                        Approve settings
+                    </PieFormLabel>
```

Comment:
I am clicking the label for the switch on the deployed NextJS app but it does not seem to visually focus or 'check' the switch

Reply:
Can you try removing the other Switch component from the DOM and trying clicking again?

Reply:
When I was testing the changes I had one component of each (input and switch) and they didn't had that issue. 🤔

Reply:
Ah I think you've got it - removing the second switch appears to resolve the issue when I test the branch locally. I wonder if it is to do with the duplicate ID attributes we spotted on the underlying input?

Reply:
I'll arrange a ticket for checking that.